src/main.cpp

```diff
@@ -216,13 +216,38 @@ void displayHelp() {
 }
 
 void destroy() {
+	char *su = getenv("SUDO_USER"); // get the username of whom runs sudo
+	if(!su) {
+		throw std::runtime_error("Please run this command via \'sudo\'");
+	}
+
+	std::cout << "Are you sure you want to " << global::color::title 
+			  << "uninstall tldr?" << global::color::dfault << " (y/n): ";
+	string yesno;
+	std::cin >> yesno;
+	if(yesno != "y") {
+		throw std::runtime_error("Canceled uninstallation");
+	}
+
 	if(unlink("/usr/bin/tldr") != 0) { // this only removes the filesystem entry, so the file will only be gone if the last descriptor to it is closed
 		throw std::runtime_error("/usr/bin/tldr could not be removed");
 	}
 	std::cout << "Removed executable /usr/bin/tldr" << std::endl;
 
-	struct passwd *pw = getpwuid(getuid());
+	string cache_dir = "/home/" + string(su) + "/.tldr";
+	if(std::filesystem::exists(cache_dir)) {
```

Comment:
Not sure we really need `std::filesystem::exists()`. If it doesn't exist `std::filesystem::remove_all` just won't do anything. In either case, the cache is gone.

Reply:
OK, I'll remove this statment.